### PR TITLE
CATO-4529: Improving validation of AC190.

### PR DESCRIPTION
**Pull request by tmikus**

### CODE REVIEW

File: src/main/scala/uk/gov/hmrc/ct/accounts/frs102/boxes/AC5052A.scala [hunk 1]
```diff
@@ -38,11 +38,13 @@ with Validators {
   override def validate(boxRetriever: Frs102AccountsBoxRetriever): Set[CtValidation] = {
     import boxRetriever._
 
+    val isMandatory = anyHaveValue(ac52(), ac53())
```
Comment (liquidarmour) [inline src/main/scala/uk/gov/hmrc/ct/accounts/frs102/boxes/AC5052A.scala]: Isn't this really valueAllowed ?

File: src/main/scala/uk/gov/hmrc/ct/accounts/frs102/boxes/AC190.scala [hunk 2]
```diff
@@ -37,7 +37,16 @@ case class AC190(value: Option[Int]) extends CtBoxIdentifier(name = "Balance at
   }
 
   def validateTotalEqualToCurrentAmount(boxRetriever: Frs102AccountsBoxRetriever)() = {
-    failIf(this.value != boxRetriever.ac76().value) {
+    val ac76 = boxRetriever.ac76()
+    val notEqualToAC76 =
```
Comment (liquidarmour) [inline src/main/scala/uk/gov/hmrc/ct/accounts/frs102/boxes/AC190.scala]: Can we do this with a pattern match?

File: src/main/scala/uk/gov/hmrc/ct/accounts/frs102/boxes/AC5052A.scala (hunk 1 above)
Comment (tmikus) [inline src/main/scala/uk/gov/hmrc/ct/accounts/frs102/boxes/AC5052A.scala]: Yes and no... You can look at it from two angles:
- AC52 or AC53 are not empty: It is mandatory
- AC52 and AC53 are empty: It must be empty

File: src/main/scala/uk/gov/hmrc/ct/accounts/frs102/boxes/AC190.scala (hunk 2 above)
Comment (tmikus) [inline src/main/scala/uk/gov/hmrc/ct/accounts/frs102/boxes/AC190.scala]: Yes we can

**Pull request merged**